### PR TITLE
Adds books and category models and tables to schema

### PR DESCRIPTION
**Pull request by rer7891**

### CODE REVIEW

File: db/migrate/20200521025059_add_missing_indexes_on_taggings.acts_as_taggable_on_engine.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# This migration comes from acts_as_taggable_on_engine (originally 6)
+if ActiveRecord.gem_version >= Gem::Version.new('5.0')
+  class AddMissingIndexesOnTaggings < ActiveRecord::Migration[4.2]; end
+else
+  class AddMissingIndexesOnTaggings < ActiveRecord::Migration; end
+end
+AddMissingIndexesOnTaggings.class_eval do
+  def change
+    add_index ActsAsTaggableOn.taggings_table, :tag_id unless index_exists? ActsAsTaggableOn.taggings_table, :tag_id
+    add_index ActsAsTaggableOn.taggings_table, :taggable_id unless index_exists? ActsAsTaggableOn.taggings_table, :taggable_id
```
Comment (linda-le1) [inline db/migrate/20200521025059_add_missing_indexes_on_taggings.acts_as_taggable_on_engine.rb:10]: Are these all autopopulated with the **rake acts_as_taggable_on_engine:install:migrations** command?

File: app/models/book.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Book < ApplicationRecord
+  validates_presence_of :title, :description, :image,
+                        :page_count, :est_reading_time
```
Comment (linda-le1) [inline app/models/book.rb:3]: Should we validate that the page count is higher than 0 as well?

**Review by linda-le1 — APPROVED** [review]
Thanks for working on this!

File: app/models/book.rb (hunk 2 above)
Comment (leiyakenney) [reply app/models/book.rb:3]: I like that idea, @linda-le1. I think it'll keep us from getting nothing-books being entered

File: db/migrate/20200521025054_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    add_index ActsAsTaggableOn.taggings_table, :tag_id
+    add_index ActsAsTaggableOn.taggings_table, [:taggable_id, :taggable_type, :context], name: 'taggings_taggable_context_idx'
+  end
+
+  def self.down
+    drop_table ActsAsTaggableOn.taggings_table
+    drop_table ActsAsTaggableOn.tags_table
+  end
+end
```
Comment (leiyakenney) [inline db/migrate/20200521025054_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb:37]: @rer7891 can we go through this gem together? This (and the other acts_as_taggable migration files) looks like it's potentially auto-generated, but I want to make sure I'm on the right track and actually know how to use the gem properly :)

File: spec/models/book_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Book, type: :model do
+  describe 'validations' do
+    it { should validate_presence_of :title }
+    it { should validate_presence_of :description }
+    it { should validate_presence_of :image }
+    it { should validate_presence_of :est_reading_time }
+    it { should validate_presence_of :page_count} 
+  end
```
Comment (leiyakenney) [inline spec/models/book_spec.rb:10]: I know we mention this earlier, but just making a note that we should update the page_count to validate > 0 here :)

File: spec/models/book_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  describe 'relationships' do
+  end
+
```
Comment (leiyakenney) [inline spec/models/book_spec.rb:14]: Just brainstorming here for when we update this: 
book can have many authors/images
book can have one title/description
book belongs to a category
Thoughts?

**Review by leiyakenney — APPROVED** [review]
Looks good to me! Thanks @rer7891 for putting these together! 
I added a few comments; a lot of them are similar to Linda's :)

**Pull request merged**